sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.New.cs

```diff
@@ -51,13 +51,7 @@ public void TransmissionFailed(bool fromStorage, int statusCode, ConnectionVars
         public void TransmissionFailed(string message, string retryDetails, string metaData) => WriteEvent(8, message, retryDetails, metaData);
 
         [Event(9, Message = "{0} has been disposed.", Level = EventLevel.Informational)]
-        public void DisposedObject(string name)
-        {
-            if (IsEnabled(EventLevel.Informational))
-            {
-                WriteEvent(9, name);
-            }
-        }
+        public void DisposedObject(string name) => WriteEvent(9, name);
```

Comment:
Why this does not have an equivalent `NonEvent` method?

Reply:
We only need the `NonEvent` for parameters with non-value types (ie Exception).

In this specific case, `DisposedObject()` is just an informational event recording that something was disposed.

Reply:
What is the reason for checking `IsEnabled`? Is it to avoid the string allocation or avoid writing to a event source when there are no listeners?

Reply:
we're not doing extra allocations inside this `Event` method so the `IsEnabled` wasn't needed.

Reply:
For example, Writing to ETW is a costlier operation, along with extra allocations we need to refrain from writing events to ETW buffer.

Reply:
Even if the calculations are not complex, there may still be some overhead involved in generating the log message or preparing the event data. By checking IsEnabled before writing to the event source, you can skip these operations altogether when logging is disabled, reducing unnecessary work and improving performance.

Reply:
Spoke offline.
We're following the example set by OpenTelemetry. That is, the `IsEnabled` check isn't needed for methods not doing any extra work or allocations. 

For example: https://github.com/open-telemetry/opentelemetry-dotnet/blob/ecd487226f65128b5decb523614c3d77c4c42289/src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs#L254-L258